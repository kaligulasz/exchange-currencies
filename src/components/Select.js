import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledSelect = styled.select`
  display: block;
  font-weight: 300;
  line-height: 1;
  color: ${props => props.theme.color.white}; 
  padding: 0.1rem 1rem;
  text-transform: uppercase;
  box-sizing: border-box;
  margin: 0;
  border: 1px solid ${props => props.theme.color.white};
  border-radius: 0.5rem;
  appearance: none;
  background: none;
  cursor: pointer;
`;

const Select = ({
  options,
  onChange,
  selected,
  ...rest
}) => (
  <StyledSelect onChange={onChange} {...rest}>
    {options.map(option => (
      <option key={option} {...(selected === option && { selected: 'selected' })}>{option}</option>
    ))}
  </StyledSelect>
);

Select.propTypes = {
  options: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  selected: PropTypes.string.isRequired,
};

export default Select;
