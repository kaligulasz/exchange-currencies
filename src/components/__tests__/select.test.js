import React from 'react';
import { shallow } from 'enzyme';

import Select from '../Select';

describe('<Select />', () => {
  it('renders select with passed options', () => {
    const component = shallow(
      <Select
        onChange={() => {}}
        options={['USD, GBP, EUR']}
        selected="USD"
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
