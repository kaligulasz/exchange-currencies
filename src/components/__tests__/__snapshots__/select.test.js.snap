// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders select with passed options 1`] = `
<Select__StyledSelect
  defaultValue="USD"
  onChange={[Function]}
>
  <option
    key="USD, GBP, EUR"
  >
    USD, GBP, EUR
  </option>
</Select__StyledSelect>
`;
